compiler-plugin-tests/src/test/java/io/ballerina/stdlib/http/compiler/OpenAPISpecGenerationTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.http.compiler;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Stream;
+
+public class OpenAPISpecGenerationTest {
+    private static final Path RESOURCE_DIRECTORY = Paths.get("src", "test", "resources", "ballerina_sources")
+            .toAbsolutePath();
+    private static final Path DISTRIBUTION_PATH = Paths.get("../", "target", "ballerina-runtime")
+            .toAbsolutePath();
+
+    @Test
+    public void testSpecGeneration() throws IOException {
+        Path projectDirPath = RESOURCE_DIRECTORY.resolve("sample_package_20");
```

Comment:
It seems like this package only has the get resources. Shall we have a service with different resource methods and parameters? We need test the post resources with payload record type with and without `@http:Payload` annotation (making sure the code modifier is engaged properly and adds the annotation if it is not there)

Reply:
Added a suitable test case by [c97d0ef](https://github.com/ballerina-platform/module-ballerina-http/pull/2492/commits/c97d0effef0df78ce422521e90cef5b553c6b44f)